CORE-12165: key-rotation messages and config

dickon opened the pull request:
👮🏻👮🏻👮🏻 !!!! DESCRIBE YOUR CHANGES HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻


# PR Checklist:

- [ ] Have you run the unit, integration and smoke tests as described [here](https://docs.corda.net/head/testing.html)?
- [ ] If you added public APIs, did you write the JavaDocs?
- [ ] If the changes are of interest to application developers, have you added them to the [changelog](https://docs.corda.net/head/changelog.html) (`/docs/source/changelog.rst`), and potentially the [release notes](https://docs.corda.net/head/release-notes.html) (`/docs/source/release-notes.rst`)?
- [ ] If you are contributing for the first time, please read the [contributor agreement](https://docs.corda.net/head/contributing.html) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://docs.corda.net/head/contributing.html#merging-the-changes-back-into-corda).

Thanks for your code, it's appreciated! :)

anien committed:
Add first version of CryptoRewrapRequest avro schema

dickon committed:
add topics for rewrap+rekey

anien committed:
Fix typo in the topic var

anien committed:
Add new request and response for a rekey operation

anien committed:
Bump version

anien committed:
Bump version

anien committed:
Rename avro schemas, update KeyRotation schemas according the design doc, rename kafka topics

anien committed:
Add new lines

anien committed:
Rename KeyRotation response to status

anien committed:
Add requestID to request schema and rename few properties

anien committed:
Add response message and add request id to request message

anien committed:
Change duration time in avro, fix indentation

anien committed:
Add response topics for rekey and rewrap

anien committed:
Fix the var names and add start and end timestamp for key rotation

anien committed:
Update timestamps in the avro schemas, status schema contains the start time now

anien committed:
Update kdoc

anien committed:
Update avro schemas and topic names for key rotation

dickon committed:
bump API Version

dickon committed:
add response topic

dickon committed:
use the string construction function

dickon committed:
add automatic rotation flag

dickon committed:
add target key aliases

vkolomeyko commented:
Do we need this change into `5.1` branch? I also see that Jira https://r3-cev.atlassian.net/browse/CORE-12165 is scheduled for C5.2.

dickon committed:
add keyType and clarify names in IndividualKeyRotationRequest

Commit:
ES-1608: 5.2 release Branch set up

jacob-scott3141 committed:
Add scans to CODEOWNERS

Commit:
Merging forward updates from release/os/5.1 to release/os/5.2 - 2023-11-13

Commit:
Update JenkinsfileMergeAutomation

Commit:
Update JenkinsfileMergeAutomation

Commit:
Merge pull request #1336 from corda/merge-release/os/5.1-release/os/5.2-2023-11-13-1

ES-1621: Merging forward updates from release/os/5.1 to release/os/5.2 - 2023-11-13

jacob-scott3141 committed:
Merge pull request #1334 from jacob-scott3141/patch-1

ES-1620: Add scans to CODEOWNERS

jacob-scott3141 committed:
Create remove-stale-prs workflow (#1337)

jacob-scott3141 committed:
CementApi and update 5.1.0 files to 5.2.0 (#1338)

nikinagy committed:
CORE-17355 - Reduce number of p2p e2e sessions maintained per pair of vnodes (#1307)

In order to increase our performance, this PR reduces number of p2p e2e sessions maintained per pair of vnodes:
for member <-> MGM communication reduce to 1
for member <-> member communication reduce to 1

jacob-scott3141 committed:
ES-1486 Roll out and activation on all C5 repos  (#1344)

* Update remove-stale-branches.yml

* Update remove-stale-prs.yml

dickon committed:
remove simulate, limit and timeToLive for now, to keep things simple until we understand the problems

dickon committed:
fix rebase error

corda-jenkins-ci02[bot] committed:
CORE-18390: Allow old version of SetOwnRegistrationStatus in the command (#1342) (#1347)

Co-authored-by: Yiftach Kaplan <67583323+yift-r3@users.noreply.github.com>
Co-authored-by: Yiftach Kaplan <yiftach.kaplan@r3.com>

dickon committed:
improve consistency

dickon committed:
add dot

dickon committed:
Merge remote-tracking branch 'origin/release/os/5.2' into feature/CORE-12165/key-rotation

# Conflicts:
#	gradle.properties

dickon committed:
Merge pull request #1350 from corda/dickon/CORE-12165/key-rotation-merge-5.2

CORE-12165: merge in 5.2 changes

dickon committed:
Merge pull request #1349 from corda/dickon/CORE-12165-feature-rotation-no-limit

CORE-12165: undo revert of revert and remove limit and time to live on key rotation

dickon commented:
duplicate
